crates/sequencer/src/sequencer.rs

```diff
@@ -1016,22 +1016,29 @@ where
         &self,
         pg_connector: PostgresConnector,
     ) -> Result<(), anyhow::Error> {
+        let db_commitment = match pg_connector.get_last_commitment().await? {
+            Some(comm) => comm,
+            None => return Ok(()),
+        };
         let ledger_commitment_l2_height = self
             .ledger_db
             .get_last_sequencer_commitment_l2_height()?
-            .ok_or(anyhow!("No commitment exists"))?;
-
-        let db_commitment = pg_connector.get_last_commitment().await?;
-        // check if last commitment in db matches sequencer's last commitment
-        if let Some(db_commitment) = db_commitment {
-            // this means that the last commitment in the db is not the same as the sequencer's last commitment
-            if db_commitment.l2_start_height > ledger_commitment_l2_height.0 {
-                self.ledger_db
-                    .set_last_sequencer_commitment_l2_height(BatchNumber(
-                        db_commitment.l2_end_height,
-                    ))?
-            }
+            .unwrap_or_default();
+        if ledger_commitment_l2_height.0 >= db_commitment.l2_end_height {
+            return Ok(());
         }
+
+        self.ledger_db
+            .set_last_sequencer_commitment_l2_height(BatchNumber(db_commitment.l2_end_height))?;
+
+        let batches = self.ledger_db.get_soft_batch_range(
+            &(BatchNumber(db_commitment.l2_start_height)
+                ..BatchNumber(db_commitment.l2_start_height + 1)),
+        )?;
+        let start_batch = &batches[0];
```

Comment:
```suggestion
        let last_soft_batch = &batches.last();
```

Reply:
If you check the range statement above, we are only getting 1 batch. Hence, it should be safe to get the value right out.